lib/spack/spack/test/concretize_preferences.py

```diff
@@ -119,6 +119,20 @@ def test_develop(self):
         spec = Spec('builtin.mock.develop-test')
         spec.concretize()
         assert spec.version == spack.spec.Version('0.2.15')
+        spec = Spec('builtin.mock.develop-test2')
+        spec.concretize()
+        assert spec.version == spack.spec.Version('0.2.15')
+
+        # now add packages.yaml with develop version
+        conf = syaml.load("""\
+dealii:
+        version: [develop]
+""")
+        spack.config.update_config('packages', conf, 'concretize')
+        # ensure that once config is in place, develop is used
+        spec = Spec('dealii')
+        spec.concretize()
```

Comment:
@tgamblin can you advise why in this unit test i get `E           UnknownPackageError: Package dealii not found.`? Is there a separate pool of packages used for testing like `mpich` (in addition to `builtin.mock.`)?